### PR TITLE
Fix #9096: sphinx-build: the value of progress bar for paralle build is wrong

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Bugfix

### Purpose
- The value of progress bar (reading and writing) should be increased just after
each task is finished.
- refs: #9096

### CODE REVIEW

File: sphinx/builders/__init__.py [hunk 1]
```diff
@@ -563,19 +570,26 @@ def write_process(docs: List[Tuple[str, nodes.document]]) -> None:
         tasks = ParallelTasks(nproc)
         chunks = make_chunks(docnames, nproc)
 
+        # create a statas_iterator to step progressbar after reading a document
+        # (see: ``on_chunk_done()`` function)
+        progress = status_iterator(chunks, __('reading sources... '), "purple",
```
Comment (peterbell10) [inline sphinx/builders/__init__.py]:
```suggestion
        # create a statas_iterator to step progressbar after writing a document
        # (see: ``on_chunk_done()`` function)
        progress = status_iterator(chunks, __('writing output... '), "darkgreen",
```

**Review by peterbell10 — COMMENTED** [review]
Small copy-paste error, otherwise this looks good and I confirmed it fixes the issue.

File: sphinx/builders/__init__.py [hunk 2]
```diff
@@ -429,6 +430,13 @@ def _read_serial(self, docnames: List[str]) -> None:
             self.read_doc(docname)
 
     def _read_parallel(self, docnames: List[str], nproc: int) -> None:
+        chunks = make_chunks(docnames, nproc)
+
+        # create a statas_iterator to step progressbar after reading a document
```
Comment (jeanas) [inline sphinx/builders/__init__.py]:
```suggestion
        # create a status_iterator to step progressbar after reading a document
```

File: sphinx/builders/__init__.py [hunk 3]
```diff
@@ -563,19 +570,26 @@ def write_process(docs: List[Tuple[str, nodes.document]]) -> None:
         tasks = ParallelTasks(nproc)
         chunks = make_chunks(docnames, nproc)
 
+        # create a statas_iterator to step progressbar after reading a document
```
Comment (jeanas) [inline sphinx/builders/__init__.py]:
```suggestion
        # create a status_iterator to step progressbar after reading a document
```

**Comment by tk0miya** [comment]
Thank you for your comments. Merging this now.

**Comment by Jean-Abou-Samra** [comment]
Thank you for the fix!

**Pull request merged**